### PR TITLE
Recognition service proto

### PR DESCRIPTION
**Pull request by maciekrb**
This is a cleaned up version for recognition proto for review. They are not building properly at the moment. 

```
In file included from /src/build/C++/recognition_service.pb.cc:5:0:
/src/build/C++/recognition_service.pb.h:132:7: error: 'const google::protobuf::RepeatedPtrField<admobilize_vision::Descriptor>& admobilize_vision::StoreDescriptorRequest::descriptor() const' cannot be overloaded
       descriptor() const;
       ^
/src/build/C++/recognition_service.pb.h:65:48: error: with 'static const google::protobuf::Descriptor* admobilize_vision::StoreDescriptorRequest::descriptor()'
   static const ::google::protobuf::Descriptor* descriptor();
```

### CODE REVIEW

File: vision/recognition_service.proto [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  // Signals the desired algorithm for recognitions
+  string matching_algorithm_version = 3;
+}
+
+// Describes descriptor matches
+message DescriptorMatch {
+  // user set tag
+  string tag = 1;
+
+  // Score for the match
```
Comment (arhuaco) [inline vision/recognition_service.proto]: Let's say that the scale ranges from 0 to 1. A bigger value means a closer match that you shouldn't make statistical interpretation of the scores.

File: vision/recognition_service.proto [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  repeated FaceDescriptor face_descriptor = 2;
+
+  // Signals the desired algorithm for recognitions
+  string matching_algorithm_version = 3;
+}
+
+// Describes descriptor matches
+message DescriptorMatch {
+  // user set tag
+  string tag = 1;
```
Comment (arhuaco) [inline vision/recognition_service.proto]: Repeated?

File: vision/recognition_service.proto [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  // Save a descriptor 
+  // The received Recognition request may:
+  //   - contain the calculated descriptors
+  //   - contain a VisionRequest with a populated ImageList field
+  //   - contain a VisionRequest with a populated Video field
+  //
+  // The tag is mandatory for a store call to be successful.
+  // When device_id items are specified, the descriptor will be scoped
+  // to the given device IDs, meaning that match requests will only
+  // succeed when correct device_id and credentials match.
```
Comment (arhuaco) [inline vision/recognition_service.proto]: Is there further validation of device_id? (That the account is the owner of the id?)

File: vision/recognition_service.proto [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  // When the save is successfull, a descriptor list with the computed
+  // descriptor data and uuid is returned
+  // The message will contain an error if the descriptor can't be stored
+  rpc StoreDescriptors(StoreDescriptorRequest) returns (StoreDescriptorResponse) {}
+
+  // Obtain descriptors saved under a user account
+  //
+  // Return values
+  // A list of descriptors according to provided uuids and device_id.
+  // if both, uuid and device_id are defined, only descriptors
```
Comment (arhuaco) [inline vision/recognition_service.proto]: "If both uuid and device_id are defined,"

File: vision/recognition_service.proto [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  // If no uuid or devices_id are defined, all descriptors available for
+  // the user account will be returned
+  // An empty list will be returned if no descriptors are found
+  rpc GetDescriptors(GetDescriptorRequest) returns (GetDescriptorResponse) {}
+
+  // Delete descriptors saved under a user account
+  // Method will delete all descriptors matching provided uuids. When
+  // a device_id is provided, only those associated with the device will
+  // be deleted
+  rpc DeleteDescriptor(DeleteDescriptorRequest) returns (DeleteDescriptorResponse) {}
```
Comment (arhuaco) [inline vision/recognition_service.proto]: device_id is not a field of DeleteDescriptorRequest

File: vision/vision.proto [hunk 6]
```diff
@@ -195,6 +195,19 @@ message Video {
...
 }
 
+// Describes a recognition descriptor
+message FaceDescriptor {
+
+  // A unique identifier for the descriptor according to UUID v4
+  string uuid = 1; 
+
+  // Algorithm used to extract the descriptor
+  string extractor_algorithm_version = 2;
```
Comment (arhuaco) [inline vision/vision.proto]: Isn't this more like a data format? (closely related to extractor algorithm...). It's the same, just wondering which is better to specify.

File: vision/recognition_service.proto [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+  // Arbitrary user defined string for the descriptor
+  string tag = 3;
+
+  // Optional device IDs to associate the descriptor
+  // with a set of devices
+  repeated string device_id = 4;
+}
+
+message StoreDescriptorResponse {
```
Comment (maciekrb) [inline vision/recognition_service.proto]: agree

File: vision/recognition_service.proto [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+// *********************** StoreDescriptor method  ***************************
+
+message StoreDescriptorRequest {
+
+  // Source images / video to obtain descriptors
+  VisionRequest vision_request = 1;
+
+  // List of descriptors
+  repeated FaceDescriptor face_descriptor = 2;
```
Comment (maciekrb) [inline vision/recognition_service.proto]: solved

File: vision/recognition_service.proto [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+ */
+
+syntax = "proto3";
+
+package admobilize_vision;
+
+import "vision.proto";
+import "vision_service.proto";
+
+// *********************** StoreDescriptor method  ***************************
```
Comment (maciekrb) [inline vision/recognition_service.proto]: agree

**Comment by maciekrb** [comment]
Comments addressed, all compiling.

File: vision/recognition_service.proto [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+  repeated Descriptor descriptors_list = 2;
+
+  // Signals the desired algorithm for recognitions
+  string matching_algorithm_version = 3;
+}
+
+// Message describes a descriptor match
+message DescriptorMatch {
+
+  // user set tags
```
Comment (arhuaco) [inline vision/recognition_service.proto]: Casing.

**Pull request merged**